pkg/ensurance/analyzer/sort/evict-sort.go

```diff
@@ -0,0 +1,182 @@
+package sort
+
+import (
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/gocrane/crane/pkg/ensurance/executor"
+	"github.com/gocrane/crane/pkg/utils"
+)
+
+// EvictRankFunc sorts the pods in eviction order
+type EvictRankFunc func(pods []executor.EvictPod)
+
+var evictSortFunc = map[string]func(p1, p2 executor.EvictPod) int{
+	"ExtResourceEnable": evictUseExtResource,
+	"ClassAndPriority":  evictClassAndPriority,
+	"ExtResourceUsage":  evictExtCpuUsed2Limit,
+	"CpuUsage":          evictCpuUsage,
+	"RunningTime":       evictStartTime,
+}
+
+func EvictRankFuncConstruct(customize []string) EvictRankFunc {
+	var evictRankFunc EvictRankFunc
+	if len(customize) == 0 {
+		cmp := []cmpEvictFunc{}
+		for _, sortFunc := range customize {
+			if f, ok := evictSortFunc[sortFunc]; ok {
+				cmp = append(cmp,f)
+			}
+			evictRankFunc = evictOrderedBy(cmp...).Sort
+			return evictRankFunc
+		}
+	}
+	evictRankFunc = evictRankCpuPressure
+
+	return evictRankFunc
+}
+
+func evictRankCpuPressure(pods []executor.EvictPod) {
+	evictOrderedBy(evictUseExtResource, evictClassAndPriority, evictExtCpuUsed2Limit, evictCpuUsage, evictStartTime).Sort(pods)
+}
+
+// evictStartTime compares pods by pod's start time
+func evictStartTime(p1, p2 executor.EvictPod) int {
+	t1 := p1.StartTime
+	t2 := p2.StartTime
+
+	if t1 == nil && t2 == nil {
+		return 0
+	}
+	if t1 == nil {
+		// maybe t1 is just starting
+		return 1
+	}
+	if t2 == nil {
+		return -1
+	}
+
+	if t1.Before(t2) {
+		return 1
+	} else if t1.Equal(t2) {
+		return 0
```

Comment:
Can it rewrite like this:

	if p1.StartTime.Equal(p2.StartTime) {
		return 0
	}else if p1.StartTime.Before(p2.StartTime) {
		return -1
	}

	return -1
        return -1 

in the Equal and Before it judge if the t1 and t2 is empty

Reply:
I rewrite it like this.
```
        t1 := p1.StartTime
	t2 := p2.StartTime

	if t1.Before(t2) {
		return 1
	} else if t1.Equal(t2) {
		return 0
	}

	if t1 == nil {
		// maybe t1 is just starting
		return -1
	}
	if t2 == nil {
		return 1
	}
	return -1
```